src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -3177,6 +3176,82 @@ protected override void HandleSetInstructionBreakpointsRequestAsync(IRequestResp
+        {
+            string expression = responder.Arguments.Expression;
+            string value = responder.Arguments.Value;
+            int frameId = responder.Arguments.FrameId.GetValueOrDefault(-1);
+
+            // if we are not stopped don't try to set
+            if (!m_isStopped)
+            {
+                responder.SetError(new ProtocolException(AD7Resources.Error_TargetNotStopped, new Message(1105, AD7Resources.Error_TargetNotStopped)));
+                return;
+            }
+
+            uint radix = GetRadixFromValueForamt(responder.Arguments.Format);
+
+            var eb = new ErrorBuilder(() => string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_SetExpression, expression, value));
+
+            try
+            {
+                // Create variable
+                GetDebugPropertyFromExpression(
+                    eb,
+                    expression,
+                    frameId,
+                    false,
+                    enum_EVALFLAGS.EVAL_RETURNVALUE | enum_EVALFLAGS.EVAL_NOEVENTS | (enum_EVALFLAGS)enum_EVALFLAGS110.EVAL110_FORCE_REAL_FUNCEVAL,
+                    DAPEvalFlags.NONE,
+                    out IDebugProperty2 property
+                );
+
+                // Assign value
+                string error = null;
+                int hr;
+                if (property is IDebugProperty3 prop3)
+                {
+                    hr = prop3.SetValueAsStringWithError(value, Constants.EvaluationRadix, Constants.EvaluationTimeout, out error);
+                }
+                else
+                {
+                    hr = property.SetValueAsString(value, Constants.EvaluationRadix, Constants.EvaluationTimeout);
+                }
+
+                if (hr != HRConstants.S_OK)
+                {
+                    string message = error ?? AD7Resources.Error_SetVariableFailed;
+                    throw new ProtocolException(message, new Message(1107, message));
+                }
+
+                // Query for new value
+                GetDebugPropertyFromExpression(
+                    eb,
+                    expression,
+                    frameId,
+                    true, // Treat SetExpression like expressions from the top frame.
+                    enum_EVALFLAGS.EVAL_RETURNVALUE | enum_EVALFLAGS.EVAL_NOEVENTS | (enum_EVALFLAGS)enum_EVALFLAGS110.EVAL110_FORCE_REAL_FUNCEVAL,
+                    DAPEvalFlags.NONE,
+                    out property
+                );
+
+                DEBUG_PROPERTY_INFO[] propertyInfo = new DEBUG_PROPERTY_INFO[1];
+                enum_DEBUGPROP_INFO_FLAGS propertyInfoFlags = GetDefaultPropertyInfoFlags();
+
+                eb.CheckHR(property.GetPropertyInfo(propertyInfoFlags, radix, Constants.EvaluationTimeout, null, 0, propertyInfo));
+
+                responder.SetResponse(new SetExpressionResponse
+                {
+                    Value = propertyInfo[0].bstrValue
```

Comment:
>Value [](http://example.com/codeflow?start=20&length=5)

If it is easy, we might want to also return `Type`, `               VariablesReference` and `MemoryReference` like we do for EvaluateResponse (line 3011).